lib/json_matchers/matcher.rb

```diff
@@ -10,16 +10,22 @@ def initialize(schema_path, **options)
     def matches?(response)
       @response = response
 
-      JSON::Validator.validate!(
-        schema_path.to_s,
-        response.body,
-        options,
-      )
-    rescue JSON::Schema::ValidationError => ex
-      @validation_failure_message = ex.message
-      false
-    rescue JSON::ParserError
-      raise InvalidSchemaError
+      begin
+        add_schemata_to_document_store
```

Comment:
Yeah, feels weird to me too. It would make sense if they were added when the matcher is defined — or, if that feels funky, then a method to load all the schemata could be implemented separate from the matcher and users using this gem could be instructed to add one method call in their RSpec configuration file.

Having the schemata added on every `#matches?` call feels very funky.

I'll fork this repo and this PR as I believe more in `json_schema` than `json-schema` and I like the direction this PR is heading. I'll try different ways of loading the schemata, to see how it feels.